File: test/translation/ts/unary.ts
let count01 = 0;

// Pure unary
count01++;

// Assignment
let count02 = count01++;

// Self-assignment
count01 = count01++;

// Self-referencing assignment
count01 += count01++;


let count03 = 0;

// Pure unary
--count03;

// Assignment
let count04 = --count03;

// Self-assignment
count03 = --count03;

// Self-referencing assignment
count03 += --count03;

File: test/translation/lua/unary.lua
local count01 = 0

count01 = count01+1
local count02 = (function()
    local count01_before = count01

    count01 = count01+1
    return count01_before
end
)()

count01 = (function()
    local count01_before = count01

    count01 = count01+1
    return count01_before
end
)()
count01 = (function()
    return count01
end
)()+(function()
    local count01_before = count01

    count01 = count01+1
    return count01_before
end
)()
local count03 = 0

count03 = count03-1
local count04 = (function()
    count03 = count03-1
    return count03
end
)()

count03 = (function()
    count03 = count03-1
    return count03
end
)()
count03 = (function()
    return count03
end
)()+(function()
    count03 = count03-1
    return count03
end
)()
